kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/DebeziumTransform.java

```diff
@@ -0,0 +1,183 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.tabular.iceberg.connect.transforms;
+
+import java.util.Map;
+import jdk.jfr.Experimental;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.data.Timestamp;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.apache.kafka.connect.transforms.util.Requirements;
+import org.apache.kafka.connect.transforms.util.SchemaUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Experimental
+public class DebeziumTransform<R extends ConnectRecord<R>> implements Transformation<R> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DebeziumTransform.class.getName());
+  private static final ConfigDef EMPTY_CONFIG = new ConfigDef();
+
+  @Override
+  public R apply(R record) {
+    if (record.value() == null) {
+      return record;
+    } else if (record.valueSchema() == null) {
+      return applySchemaless(record);
+    } else {
+      return applyWithSchema(record);
+    }
+  }
+
+  private R applyWithSchema(R record) {
+    Struct value = Requirements.requireStruct(record.value(), "Debezium transform");
+
+    String op = mapOperation(value.get("op").toString());
+
+    Struct payload;
+    Schema payloadSchema;
+    if (op.equals(CdcConstants.OP_DELETE)) {
+      payload = value.getStruct("before");
+      payloadSchema = value.schema().field("before").schema();
+    } else {
+      payload = value.getStruct("after");
+      payloadSchema = value.schema().field("after").schema();
+    }
+
+    Schema newValueSchema = makeUpdatedSchema(payloadSchema);
+    Struct newValue = new Struct(newValueSchema);
+
+    for (Field field : payloadSchema.fields()) {
+      newValue.put(field.name(), payload.get(field));
+    }
+    newValue.put(CdcConstants.COL_CDC_OP, op);
+    newValue.put(CdcConstants.COL_CDC_TS, new java.util.Date(value.getInt64("ts_ms")));
+    newValue.put(CdcConstants.COL_CDC_TABLE, tableNameFromSourceStruct(value.getStruct("source")));
+
+    return record.newRecord(
+        record.topic(),
+        record.kafkaPartition(),
+        record.keySchema(),
+        record.key(),
+        newValueSchema,
+        newValue,
+        record.timestamp());
+  }
+
+  @SuppressWarnings("unchecked")
+  private R applySchemaless(R record) {
+    Map<String, Object> value = Requirements.requireMap(record.value(), "Debezium transform");
+
+    String op = mapOperation(value.get("op").toString());
+
+    Object payload;
+    if (op.equals(CdcConstants.OP_DELETE)) {
```

Comment:
I was just looking at some update examples that appear to have both `before` and `after`, but this doesn't look like it would support that.  Just wonder if there's a case for both.

Reply:
For inserts or updates we want `after`. For deletes, after will be null, so we pick `before`. Did I understand the question right?